File: dotnet/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/Factory/Concrete/LinuxFactory.cs
using AbstractFactoryDesignPattern.Example1.Factory.Abstract;
using AbstractFactoryDesignPattern.Example1.Models.Abstract;
using AbstractFactoryDesignPattern.Example1.Models.Concrete;

namespace AbstractFactoryDesignPattern.Example1.Factory.Concrete;

public class LinuxFactory : GUIFactory
{
	public override Button CreateButton()
	{
		return new LinuxButton();
	}
	public override Menu CreateMenu()
	{
		return new LinuxMenu();
	}
	public override Dialog CreateDialog()
	{
		return new LinuxDialog();
	}
}

File: nodejs/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/src/factory/concrete/LinuxFactory.ts
import { GUIFactory } from '../abstract/GUIFactory';
import { Button } from '../../models/abstract/Button';
import { Menu } from '../../models/abstract/Menu';
import { Dialog } from '../../models/abstract/Dialog';
import { LinuxButton } from '../../models/concrete/LinuxButton';
import { LinuxMenu } from '../../models/concrete/LinuxMenu';
import { LinuxDialog } from '../../models/concrete/LinuxDialog';

export class LinuxFactory extends GUIFactory {
  createButton(): Button { return new LinuxButton(); }
  createMenu(): Menu { return new LinuxMenu(); }
  createDialog(): Dialog { return new LinuxDialog(); }
}
